### PR TITLE
Remove nested steps JSON printing due issues with crawler

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/reporters/ConcurrentStoryReporter.java [hunk 1]
```diff
@@ -92,7 +92,7 @@
         }
     }
 
-    private List<DelayedMethod> delayedMethods = new ArrayList<>();
+    private List<DelayedMethod> delayedMethods = Collections.synchronizedList(new ArrayList<DelayedMethod>());
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/ConcurrentStoryReporter.java]: What's the reason of this change? It may significantly decrease performance

File: jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java [hunk 2]
```diff
@@ -50,6 +51,9 @@
     private boolean scenarioCompleted = false;
     private boolean stepPublishing = false;
 
+    private boolean mainStepStarted = false;
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]: isn't it equal to `subStepsLevel == 0`?

File: jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java [hunk 3]
```diff
@@ -70,56 +74,78 @@ protected void print(PrintStream output, String text) {
 
     @Override
     public void beforeStep(String step) {
-        printSubSteps();
-        super.beforeStep(step);
+        if(mainStepStarted) {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]: keep formatting: `if (..`

Comment (valfirst) [reply jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]: here and everywhere below

File: jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java [hunk 4]
```diff
@@ -70,56 +74,78 @@ protected void print(PrintStream output, String text) {
...
+            subStepsLevel++;
+        } else {
+            mainStepStarted = true;
+            expectedLevelToPrint.set(subStepsLevel + 1);
+            printSubSteps();
+            super.beforeStep(step);
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]: why isn't it invoked for sub-steps?

File: jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java [hunk 5]
```diff
@@ -276,6 +302,18 @@ private static Properties defaultXmlPatterns() {
         return patterns;
     }
 
+    private boolean checkForPrint() {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]: method name is not clear

File: jbehave-core/src/test/java/org/jbehave/core/reporters/PrintStreamOutputBehaviour.java [hunk 6]
```diff
@@ -20,18 +45,6 @@
...
-import java.util.Locale;
-import java.util.Properties;
-
-import static java.util.Arrays.asList;
-import static org.hamcrest.CoreMatchers.startsWith;
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.*;
-import static org.jbehave.core.reporters.Format.HTML;
-import static org.jbehave.core.reporters.Format.TXT;
-
```
Comment (valfirst) [inline jbehave-core/src/test/java/org/jbehave/core/reporters/PrintStreamOutputBehaviour.java]: a lot of noisy changes: keep original imports order

File: jbehave-core/src/main/java/org/jbehave/core/reporters/ConcurrentStoryReporter.java (hunk 1 above)
Comment (uarlouski) [reply jbehave-core/src/main/java/org/jbehave/core/reporters/ConcurrentStoryReporter.java]: If it's not synchronized the number of beforeStep invocations and sucessfull/failed invocation is not equal

File: jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java [hunk 7]
```diff
@@ -277,6 +299,18 @@ private static Properties defaultXmlPatterns() {
         return patterns;
     }
 
+    private boolean ifMainStepAllowPrintOrElseReduceLevel() {
+        if (expectedLevelToPrint == 0) {
+            return true;
+        } else if (subStepsLevel.get() == expectedLevelToPrint) {
+            expectedLevelToPrint = 0;
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]: please explain this logic

File: jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java [hunk 8]
```diff
@@ -277,6 +297,15 @@ private static Properties defaultXmlPatterns() {
...
     }
 
+    private boolean ifMainStepAllowPrintOrElseReduceLevel() {
+        if (subStepsLevel.get() <= 1) {
+            return true;
+        } else {
+            subStepsLevel.decrementAndGet();
+            return false;
+        }
+    }
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]:
```java
private boolean isFirstStepLevel() {
    subStepsLevel.get() <= 1;
}
```

File: jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java [hunk 9]
```diff
@@ -277,6 +297,15 @@ private static Properties defaultXmlPatterns() {
         return patterns;
     }
 
+    private boolean ifMainStepAllowPrintOrElseReduceLevel() {
+        if (subStepsLevel.get() <= 1) {
+            return true;
+        } else {
+            subStepsLevel.decrementAndGet();
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]: this is a side-effect of the method
